atom/browser/extensions/tab_helper.cc

```diff
@@ -132,6 +133,15 @@ void TabHelper::CreateTab(content::WebContents* owner,
         profile->original_context()->partition_with_prefix());
   }
 
+  params->GetString("src", &newTabURL);
```

Comment:
as discussed this will not catch all types of navigations

Reply:
Will check chromium code base to see how this is addressed.